spec/models/provider_spec.rb

```diff
@@ -61,4 +61,23 @@
       expect(tenant.providers).to include(provider)
     end
   end
+
+  context "#destroy_queue" do
+    it "destroy when has no managers" do
+      expect(provider).to receive(:destroy)
+      provider.destroy_queue
+    end
```

Comment:
What?  If we are asking the provider to destroy something over the queue, then we would expect the provider to_**not** receive destroy.

Reply:
this is when there's NO manager. Provider will be destroyed.

Probably I can add an explicit `provider.managers  =[]` to make it more obvious.